src/transaction.ts

```diff
@@ -1975,69 +2074,77 @@ export class Transaction extends Dml {
     const requestOptions = (options as CommitOptions).requestOptions;
     const reqOpts: CommitRequest = {mutations, session, requestOptions};
 
-    const span = getActiveOrNoopSpan();
-
-    if (this.id) {
-      reqOpts.transactionId = this.id as Uint8Array;
-    } else if (!this._useInRunner) {
-      reqOpts.singleUseTransaction = this._options;
-    } else {
-      this.begin().then(() => this.commit(options, callback), callback);
-      return;
-    }
-
-    if (
-      'returnCommitStats' in options &&
-      (options as CommitOptions).returnCommitStats
-    ) {
-      reqOpts.returnCommitStats = (options as CommitOptions).returnCommitStats;
-    }
-    if (
-      'maxCommitDelay' in options &&
-      (options as CommitOptions).maxCommitDelay
-    ) {
-      reqOpts.maxCommitDelay = (options as CommitOptions).maxCommitDelay;
-    }
-    reqOpts.requestOptions = Object.assign(
-      requestOptions || {},
-      this.requestOptions
-    );
+    const q = {opts: this._observabilityOptions};
+    return startTrace('Transaction.commit', q, span => {
+      if (this.id) {
+        reqOpts.transactionId = this.id as Uint8Array;
+      } else if (!this._useInRunner) {
+        reqOpts.singleUseTransaction = this._options;
+      } else {
+        this.begin().then(() => {
+          span.end();
```

Comment:
this span.end() should be after `commit` that is how we will have nested spans for commit

Reply:
Good catch and thank you!

Reply:
@odeke-em What about this comment ?

Reply:
I did address it by invoking span.end() after this.commit was invoked.